python/seldon_core/wrapper.py

```diff
@@ -126,10 +128,15 @@ def HealthStatus():
 
     @app.route("/metadata", methods=["GET"])
     def Metadata():
+        if metadata_data is None:
+            raise SeldonMicroserviceException(
```

Comment:
We probably want to just return empty metadata if it's not defined, instead of 500, as there's not really an internal error if metadata is not defined

Reply:
That's the behaviour, the implementation may be a bit odd though.
Basically `metadata_data` is `None` when there was an error validating it. 

It is empty metadata when user didn't define anything.

Reply:
I see, but do we want to throw a server error every time the endpoint is called? Probably best woudl be to throw the error when starting up and validating the metadata instead of waiting until the endpoint is called

Reply:
This is the same as the `metrics` endpoint - it will raise SeldonMicroserviceException in case of bad metrics returned by `metrics` method whenever the endpoint is called.

There during initial metadata validation I will log the errors but continue running in order to not stop serving inferences at `/predict` just in case of wrongly defined metadata but return 500 if someone requests `/metadata`. 

I believe that is what @cliveseldon was suggesting.

Reply:
As discussed we wouldn't want to crash microservice when metadata is incorrect - leaving as 500 for now on `/metadata` endpoint.